src/osd/osd_types.h

```diff
@@ -523,7 +530,8 @@ class coll_t {
   spg_t pgid;
   uint64_t removal_seq;  // note: deprecated, not encoded
 
-  string _str;  // cached string
+  char _str_buff[spg_t::calc_name_buf_size];
+  char *_str = NULL;
```

Comment:
This means we need an operator=.  Otherwise we'll copy the pointer value and bad things will happen